events/api.py

```diff
@@ -2452,8 +2465,33 @@ def filter_queryset(self, queryset):
     def allow_bulk_destroy(self, qs, filtered):
         return False
 
-    def update(self, *args, **kwargs):
-        response = super().update(*args, **kwargs)
+    def update(self, request, *args, **kwargs):
+        pk = kwargs.get('pk', None)
+        queryset = Event.objects.filter(id=pk).prefetch_related('offers', 'images__publisher', 'external_links',
+                                                                'sub_events', 'in_language', 'videos'
+                                                                ).select_related('publisher')
+        context = self.get_serializer_context()
+
+        context['queryset'] = queryset
+        context['keywords'], context['location'], bulk = self.cache_related_fields(request)
+        serializer = EventSerializer(Event.objects.get(id=pk), data=request.data, context=context)
+        serializer.is_valid(raise_exception=True)
+
+        if not self.request.user.can_edit_event(serializer.instance.publisher, serializer.instance.publication_status):
+            raise DRFPermissionDenied()
+        if isinstance(serializer.validated_data, list):
+            event_data_list = serializer.validated_data
+        else:
+            event_data_list = [serializer.validated_data]
+        for event_data in event_data_list:
+            org = self.organization
+            if hasattr(event_data, 'publisher'):
+                org = event_data['publisher']
+            if not self.request.user.can_edit_event(org, event_data['publication_status']):
+                raise DRFPermissionDenied()
```

Comment:
This could have been in different commit so 1 commit contains changes related to 1 subject only.